LaTeX/TeX.sublime-syntax

```diff
@@ -7,73 +7,181 @@ file_extensions:
   - cls
 scope: text.tex
 contexts:
+  prototype:
+    - include: comments
+
   main:
-    - match: (\\)(backmatter|else|fi|frontmatter|ftrue|mainmatter|if(case|cat|dim|eof|false|hbox|hmode|inner|mmode|num|odd|undefined|vbox|vmode|void|x)?)\b
-      scope: keyword.control.tex
-      captures:
-        1: punctuation.definition.keyword.tex
-    - match: ((\\)catcode)`(?:\\)?.(=)(\d+)
-      scope: meta.catcode.tex
+    - include: macros
+    - include: controls
+    - include: catcode
+    - include: braces
+    - include: boxes
+    - include: block-math
+    - include: inline-math
+    - include: general-constants
+    - include: general-commands
+
+  general-commands:
+    - match: '(\\)[A-Za-z@]+'
+      scope: support.function.general.tex
       captures:
-        1: keyword.control.catcode.tex
-        2: punctuation.definition.keyword.tex
-        3: punctuation.separator.key-value.tex
-        4: constant.numeric.category.tex
-    - match: (%:).*$\n?
-      scope: comment.line.percentage.semicolon.texshop.tex
+        1: punctuation.definition.backslash.tex
+
+  general-constants:
+    - match: \\\\
+      scope: constant.character.newline.tex
+    - match: '(\\)[^A-Za-z@]'
+      scope: constant.character.escape.tex
       captures:
-        1: punctuation.definition.comment.tex
-    - match: ^%!TEX (\S*) =\s*(.*)\s*$
-      scope: comment.line.percentage.directive.texshop.tex
-    - match: (%).*$\n?
+        1: punctuation.definition.backslash.tex
+    - match: '&'
+      scope: constant.character.ampersand.tex
+
+  comments:
+    - match: '%.*$\n?'
       scope: comment.line.percentage.tex
+
+  controls:
+    - match: ((\\)(else|fi|ftrue|if(case|cat|dim|eof|false|hbox|hmode|inner|mmode|num|odd|undefined|vbox|vmode|void|x)?))\b
       captures:
-        1: punctuation.definition.comment.tex
-    - match: '\{'
+        1: keyword.control.tex
+        2: punctuation.definition.backslash.tex
+    - match: '((\\)(?:input))'
```

Comment:
This also matches `\inputminted`, we might need to check that it is followed by a brace.